transmogrifier/sources/transformer.py

```diff
@@ -242,30 +238,21 @@ def _transform(
             source_record: A single source record.
         """
         if self.record_is_deleted(source_record):
-            source_record_id = self.get_source_record_id(source_record)
-            timdex_record_id = self.get_timdex_record_id(
-                self.source, source_record_id, source_record
-            )
+            timdex_record_id = self.get_timdex_record_id(source_record)
```

Comment:
Since it was brought up in an earlier discussion, do we still want a `_transform` method that is called by the other base transformers with accurate type hints or is it worth consolidating around a single method for both?

Reply:
I'm still torn about this, but I think it'd be nice to collapse to a single method on the base `Transformer` class.  Seems like if we did `source_record: Tag | dict[str, JSON]`, then each time we do `source_record` in the signatures of the field methods for the actual transforms, they could just set the correct type?

Reply:
Hmm, just to clarify-- @ehanson8 , are you asking about the `get_timdex_record_id` field method?

From my understanding, what we currently do is:
* `Transformer.get_timdex_record_id(source_record: Tag | dict[str, JSON])`
* `JSONTransformer.get_timdex_record_id(source_record: dict[str, JSON])`
* `XMLTransformer.get_timdex_record_id(source_record: Tag)`

For me, I think this is understandable!

Reply:
@ghukill @jonavellecuerdo Surfacing this thread in this increasingly labyrinthian draft PR, do we want to go with a single `Transformer.transform` method or continue with the `Transformer._transform` that is called by the `JSONTransformer` and `XMLTransformer` with more specific type hints.

I think `Transformer.transform` is preferable but defer to the majority